docs/v3/how-to-guides/workflows/assets.mdx

```diff
@@ -124,6 +124,64 @@ def generate_reports():
     create_detailed_report(data)
 ```
 
+## Dynamic asset materialization
+
+Use the `with_options` pattern to dynamically change asset keys and dependencies at runtime. This is useful when you need to parameterize asset paths based on flow inputs, dates, or other runtime conditions:
```

Comment:
maybe link to task docs on "the `with_options` pattern"?

Reply:
Linked to the API ref ('cause it looks great now!) and also updated the formatting issues there.